File: php/ExampleClient.php
<?php
include_once 'INC_COMM.php';
include_once 'MtExampleService.php';
include_once 'Types.php';

use Thrift\Transport\TBufferedTransport;
use Thrift\Transport\TSocket;
use Thrift\Protocol\TBinaryProtocol;


$ip = "localhost";
$port = 9090;
$timeout = 1000;

$socket_ = new TSocket($ip, $port);
$socket_ ->setRecvTimeout($timeout);

$bufferedSocket_ = new TBufferedTransport($socket_, 1024*1024, 1024*1024);

$protocol_ = new TBinaryProtocol($bufferedSocket_);
$bufferedSocket_->open();

$client = new MtExampleServiceClient($protocol_);

$client->GetUserProfile(0);

?>

File: python/ExampleClient.py
# coding=utf-8
import sys
sys.path.append('./') 

from thrift import Thrift
from thrift.transport import TSocket
from thrift.transport import TTransport
from thrift.protocol import TBinaryProtocol
from thrift.protocol import TCompactProtocol
from example import MtExampleService, ttypes 

def ExampleClient():
    try:
        transport = TSocket.TSocket('localhost', 9090) 
        transport = TTransport.TBufferedTransport(transport)
        protocol = TCompactProtocol.TCompactProtocol(transport)
        client = MtExampleService.Client(protocol)
        transport.open()

        client.GetUserProfile(100)

        transport.close()
    except Thrift.TException, tx:
        print '%s' % (tx.message)
        
        
if __name__ == '__main__':
    ExampleClient()
